plaso/parsers/viminfo.py

```diff
@@ -0,0 +1,525 @@
+    items = []
+    for index, item in enumerate(self.parsed_data.get(
+        'debug_line_history_items', [])):
+      items.append(
+          {'index': index, 'timestamp': int(item[5]), 'debug line': item[1]})
+    return items
+
+  def Registers(self):
+    """Returns a list of register items."""
+    items = []
+    for item in self.parsed_data.get('registers_items', []):
+      items.append({
+          'register': item[1],
+          'timestamp': int(item[12]),
+          'register_value': '\n'.join(item[4])})
+    return items
+
+  def Filemarks(self):
+    """Returns a list of filemark items."""
+    items = []
+    for index, item in enumerate(self.parsed_data.get('filemarks_items', [])):
+      items.append(
+          {'index': index, 'timestamp': int(item[5][5]), 'filename': item[4]})
+    return items
+
+  def Jumplist(self):
+    """Returns a list of jumplist items."""
+    items = []
+    for index, item in enumerate(self.parsed_data.get('jumplist_items', [])):
+      items.append(
+          {'index': index, 'timestamp': int(item[4][5]), 'filename': item[3]})
+    return items
+
+
+class VimInfoEventData(events.EventData):
+  """VimInfo event data.
+
+  Attributes:
+    type (str): the Vim history type.
+    value (str): the Vim history value.
+    filename (str): the name of the file that was opened/edited.
+    item_number (int): the nth item of the history type.
+  """
+
+  DATA_TYPE = 'viminfo:history'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(VimInfoEventData, self).__init__(data_type=self.DATA_TYPE)
+    self.type = None
+    self.value = None
+    self.filename = None
+    self.item_number = None
+
+
+class VimInfoParser(interface.FileObjectParser):
+  """Parses events from Viminfo files."""
+
+  NAME = 'viminfo'
+  DATA_FORMAT = 'Viminfo file'
+
+  _ENCODING = 'utf-8'
+
+  # 10 MiB is the maximum supported viminfo file size.
+  _MAXIMUM_VIMINFO_FILE_SIZE = 10 * 1024 * 1024
+
+  _FILENAME = '.viminfo'
+
+  @classmethod
+  def GetFormatSpecification(cls):
+    """Retrieves the format specification.
+
+    Returns:
+      FormatSpecification: a format specification or None if not available.
+    """
+    format_specification = specification.FormatSpecification(cls.NAME)
+    format_specification.AddNewSignature(
+        b'# This viminfo file was generated by Vim', offset=0)
+    return format_specification
+
+  def ParseFileObject(self, parser_mediator, file_object):
+    """Parses a plist file-like object.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      file_object (dfvfs.FileIO): a file-like object.
+
+    Raises:
+      WrongParser: when the file cannot be parsed.
+    """
+    filename = parser_mediator.GetFilename()
+    if filename != self._FILENAME:
+      raise errors.WrongParser(
+          'File name: {0:s} does not match the expected viminfo '
```

Comment:
Use bounding ( .. ) for a multi line string

Reply:
Done